src/mame/sinclair/tsconf.cpp

```diff
@@ -77,27 +76,32 @@ void tsconf_state::tsconf_mem(address_map &map)
 }
 
 void tsconf_state::tsconf_io(address_map &map)
+{
+	map(0x0000, 0xffff).lrw8(
+		NAME([this](offs_t offset) { return m_ioext.read_byte((m_beta->dos_io_r() << 16) | offset); }),
+		NAME([this](offs_t offset, u8 data) { m_ioext.write_byte((m_beta->dos_io_r() << 16) | offset, data); }));
+}
```

Comment:
Can you find a way to do this with a `memory_view`, rather than effectively creating an `address_map_bank_device` in all but name?

Reply:
That is mainly to support configuration through zxbus. I took this approach previously because zxbus is required to handle 17 bit IO where all 3 combinations are available: shadow only, only !shadow, and both.
It's not clear if such case is possible to implement using banks or memviews.

Reply:
Could you have a view with two entries, one for shadow and one for !shadow, then have devices install to one or both of the entries?  Are there actually cases where something uses the 17th bit as an address bit rather than just a select bit?

Reply:
Sorry, it still not clear for me how can we pull desired view in zxbus when installing device in IO space?
https://github.com/mamedev/mame/blob/c5c18833c896afedb52ab9e79a85dfe1dad1a99b/src/devices/bus/spectrum/zxbus.h#L91

Reply:
I've switched this PR to the views. I'll revisit zxbus mapping later as it's not critical for this work.